File: Yandex/Practicum/Algorightms_and_Data_Structures/6_sprint/B._Перевести_список_ребер_в_матрицу_смежности/solution.cpp
#include <iostream>
#include <vector>


void solution() {
    /*
    Time Complexity O(V^2)
    Memory Complexity O(V^2)
    */
    int v, e;
    std::cin >> v >> e;
    std::vector<std::vector<int>> adjMatrix (v + 1, std::vector<int>(v + 1, 0));
    for (int i = 0; i < e; ++i) {
        int source, destination;
        std::cin >> source >> destination;
        adjMatrix[source][destination] = 1;
    }
    for (int source = 1; source <= v; ++source) {
        for (int destination = 1; destination <= v; ++destination) {
            if (destination > 1) std::cout << " ";
            std::cout << adjMatrix[source][destination];
        }
        std::cout << "\n";
    }
}


int main() {
    solution();
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/6_sprint/B._Перевести_список_ребер_в_матрицу_смежности/solution.py
import sys


def solution() -> None:
    """
    Time Complexity O(V^2)
    Memory Complexity O(V^2)
    """
    v, e = map(int, sys.stdin.readline().rstrip().split())
    adj_matrix: list[list[int]] = [[0] * (v + 1) for _ in range(v + 1)]
    for _ in range(e):
        source, destination = map(int, sys.stdin.readline().rstrip().split())
        adj_matrix[source][destination] = 1
    for source in range(1, v + 1):
        for destination in range(1, v + 1):
            if destination > 1: sys.stdout.write(' ')
            sys.stdout.write(str(adj_matrix[source][destination]))
        sys.stdout.write('\n')


if __name__ == '__main__':
    solution()